### PR TITLE
fix issue with get_region_io_fds when fd = -1

### PR DESCRIPTION
**Pull request by jlevon**

### CODE REVIEW

File: lib/libvfio-user.c [hunk 1]
```diff
@@ -558,7 +558,7 @@ handle_device_get_region_io_fds(vfu_ctx_t *vfu_ctx, vfu_msg_t *msg)
     assert(msg != NULL);
     assert(msg->out_fds == NULL);
 
-    if (msg->in_size != sizeof(vfio_user_region_io_fds_request_t)) {
+    if (msg->in_size < sizeof(vfio_user_region_io_fds_request_t)) {
```
Comment (tmakatos) [inline lib/libvfio-user.c:561]: Is this related to this issue?

File: lib/libvfio-user.c [hunk 2]
```diff
@@ -576,10 +576,6 @@ handle_device_get_region_io_fds(vfu_ctx_t *vfu_ctx, vfu_msg_t *msg)
 
     vfu_reg = &vfu_ctx->reg_info[req->index];
 
-    if (vfu_reg->fd == -1) {
```
Comment (tmakatos) [inline lib/libvfio-user.c:579]: I'm lacking some context, IIUC the problem is that we don't allow a region to have ioregionfd sub-regions if the region is not memory mappable? ioregionfd and memory-mappable region are orthogonal, right?

Comment (jlevon) [reply lib/libvfio-user.c:579]: right, exactly.

File: lib/libvfio-user.c (hunk 1 above)
Comment (tmakatos) [reply lib/libvfio-user.c:561]: no it's fine, just wanted to make sure I'm not missing something

File: lib/libvfio-user.c (hunk 2 above)
Comment (tmakatos) [reply lib/libvfio-user.c:579]: Could we document that on the commit message?

**Review by tmakatos — APPROVED** [review]

File: lib/libvfio-user.c (hunk 2 above)
Comment (jlevon) [reply lib/libvfio-user.c:579]: I tried to clarify the commit message a bit more?

Comment (tmakatos) [reply lib/libvfio-user.c:579]: You did, I only looked at the PR title.. my bad.
